types/npm-packlist/index.d.ts

```diff
@@ -1,18 +1,22 @@
 // Type definitions for npm-packlist 3.0
 // Project: https://github.com/npm/npm-packlist, https://www.npmjs.com/package/npm-packlist
 // Definitions by: Klaus Meinhardt <https://github.com/ajafff>
+//                 Florian Imdahl <https://github.com/ffflorian>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.1
+// Minimum TypeScript Version: 4.3
 
-declare function packlist(options?: packlist.Options): Promise<string[]>;
-declare function packlist<T>(options: packlist.Options | undefined, callback: (result: string[]) => T): Promise<T>;
+import * as Arborist from '@npmcli/arborist';
```

Comment:
how this can be a dependency?
It's listed in dev only:
https://github.com/npm/npm-packlist/blob/main/package.json#L19

Reply:
The user is supposed to import it themselves, see https://github.com/npm/npm-packlist/blob/main/README.md?plain=1#L9. The `packlist` function still expects a `Node` from arborist and thus we need to import the types here.